### PR TITLE
Developer guide updates

### PR DESCRIPTION
**Pull request by RyanAngJY**

### CODE REVIEW

**Comment by RyanAngJY** [comment]
bold + grey box: actual commands
italics + grey box: classes
grey box: other general stuff

File: docs/DeveloperGuide.adoc [hunk 1]
```diff
@@ -127,52 +128,53 @@ Given below is a quick overview of each component:
...
 The sections below give more details of each component.
 
 [[Design-Main]]
 === Main component
-`Main` has only one class called link:{repoURL}/src/main/java/seedu/recipe/MainApp.java[`MainApp`].
+<<Design-Main,*`Main`*>> has only one class called link:{repoURL}/src/main/java/seedu/recipe/MainApp.java[`MainApp`].
 
-* At app launch: It initializes the components in the correct sequence, and connects them up with each other.
-* At shut down: It shuts down the components and invokes cleanup method where necessary.
+* At app launch: Initializes the components in the correct sequence, and connects them up with each other.
```
Comment (kokonguyen191) [inline docs/DeveloperGuide.adoc]: No subject? How about changing line 162 to
+<<Design-Main,*`Main`*>> has only one class called link:{repoURL}/src/main/java/seedu/recipe/MainApp.java[`MainApp`]. This class

File: docs/DeveloperGuide.adoc [hunk 2]
```diff
@@ -217,9 +219,9 @@ image::LogicCommandClassDiagram.png[width="800"]
...
 link:{repoURL}/src/main/java/seedu/recipe/logic/Logic.java[`Logic.java`]
 
-Refer to the Sequence Diagram for the example of interactions within the `Logic` component for the `execute("delete 1")` API call.
+Refer to the Sequence Diagram for the example of interactions within the <<Design-Logic,*`Logic`*>> component for the `execute("delete 1")` API call.
 
-.Interactions Inside the Logic Component for the `delete 1` Command
+.Interactions Inside the Logic Component for the *`delete 1`* Command
 image::DeleteRecipeSdForLogic.png[width="800"]
 
 [[Design-Model]]
```
Comment (kokonguyen191) [inline docs/DeveloperGuide.adoc]: Still needs some text here

File: docs/DeveloperGuide.adoc [hunk 3]
```diff
@@ -230,11 +232,11 @@ image::ModelClassDiagram.png[width="800"]
...
+The <<Design-Model,*`Model`*>>,
 
-* stores a `UserPref` object that represents the user's preferences.
+* stores a `_UserPref_` object that represents the user's preferences.
 * stores the Recipe Book data.
-* exposes an unmodifiable `ObservableList<Recipe>` that can be 'observed' e.g. the UI can be bound to this list so that the UI automatically updates when the data in the list change.
+* exposes an unmodifiable `_ObservableList<Recipe>_` that can be 'observed' e.g. the UI can be bound to this list so that the UI automatically updates when the data in the list change.
 * does not depend on any of the other three components.
 
 [[Design-Storage]]
```
Comment (kokonguyen191) [inline docs/DeveloperGuide.adoc]: Still needs some text here

**Pull request merged**